javascript/extractor/lib/typescript/src/common.ts

```diff
@@ -27,4 +50,74 @@ export class Project {
     this.unload();
     this.load();
   }
+
+  /**
+   * Override for module resolution in the TypeScript compiler host.
+   */
+  private resolveModuleNames(
+        moduleNames: string[],
+        containingFile: string,
+        reusedNames: string[],
+        redirectedReference: ts.ResolvedProjectReference,
+        options: ts.CompilerOptions) {
+
+    const { host, resolutionCache } = this;
+    return moduleNames.map((moduleName) => {
+      let redirected = this.redirectModuleName(moduleName, containingFile, options);
+      if (redirected != null) return redirected;
+      return ts.resolveModuleName(moduleName, containingFile, options,  host, resolutionCache).resolvedModule;
```

Comment:
Yes, I'm pretty sure the cache is just for performance.

The cache makes sense, because it's a lot of work to locate a package in the general case. The `node_modules` folder can be in a bunch of places, and there's both the actual package and its companion in `@types` to look for, and TypeScript also have to parse the `package.json` file to look at its `typings` field, and I'm sure there's more.

In our case, the `packageEntryPoints` map makes it a lot easier though, as it mostly does what the cache would do anyway. We do have to a try a few file extensions when looking for the file on disk, which I suppose we could cache as well, but I'm not sure it matters.